importlib_metadata/__init__.py

```diff
@@ -475,9 +475,18 @@ def make_file(name, hash=None, size_str=None):
 
         @pass_none
         def make_files(lines):
-            return list(starmap(make_file, csv.reader(lines)))
-
-        return make_files(self._read_files_distinfo() or self._read_files_egginfo())
+            return list(
+                filter(
+                    lambda package_path: package_path.locate().exists(),
```

Comment:
Given the complexity, I wonder if it makes sense to separate this 'extant files' behavior from the rest.

Reply:
Do you mean a simple refactoring like this?
```python
        @pass_none
        def make_files(lines):
            return list(starmap(make_file, csv.reader(lines)))

        @pass_none
        def skip_missing_files(package_paths):
            return list(filter(lambda path: path.locate().exists(), package_paths))

        return skip_missing_files(
            make_files(
                self._read_files_distinfo()
                or self._read_files_egginfo_installed()
                or self._read_files_egginfo_sources()
            )
        )
```

Reply:
Did this in https://github.com/python/importlib_metadata/pull/437/commits/9b165a91af49bb11b7cf06b5661801f603aaeba6, can revisit if needed.